code/game/objects/items/stunbaton.dm

```diff
@@ -202,12 +202,13 @@
 	hitcost = 2000
 	throw_hit_chance = 10
 	slot_flags = ITEM_SLOT_BACK
-	var/obj/item/assembly/igniter/sparkler = 0
+	var/obj/item/assembly/igniter/sparkler
 
 /obj/item/melee/baton/cattleprod/Initialize()
 	. = ..()
 	sparkler = new (src)
+	sparkler.activate_cooldown = 5
 
 /obj/item/melee/baton/cattleprod/baton_stun()
-	if(sparkler.activate())
-		..()
+	sparkler?.activate()
+	return ..()

```

Comment:
```suggestion
	. = ..()
```

Reply:
Personally I see little reason to do `. = ..()` than `return ..()` since no action follows up except the implied end-of-proc return.
Doing this should merely adds up another step to the proc, but either ways, it has virtually almost no impact to its processing time.
But I may have just said a load of ballocks, who knows...